Examples/Algorithms/TrackFinding/src/AmbiguityResolutionMLAlgorithm.cpp

```diff
@@ -0,0 +1,193 @@
+                             .index();
+          hits.emplace_back(indexHit);
+          ++nbMeasurements;
+        }
+      });
+      if (nbMeasurements < m_cfg.nMeasurementsMin) {
+        continue;
+      }
+      trackMap.emplace(nbMeasurements, std::make_pair(trackID, hits));
+      auto param = traj.trackParameters(tip);
+      trackParameters.emplace_back(param);
+      trackTips.emplace_back(iTraj, tip);
+      trackID++;
+    }
+    iTraj++;
+  }
+  // Performe the share hit based clustering
+  auto clusters = ClusteriseTrack(trackMap);
+  Acts::NetworkBatchInput networkInput(trackID + 1, 8);
+  trackID = 0;
+  // Get the input feature of the network for all the tracks
+  for (auto const& [key, val] : clusters) {
+    for (auto const& track : val) {
+      std::pair<int, int> tips = trackTips.at(track);
+      TrackParameters parameters = trackParameters.at(track);
+      auto trajState = Acts::MultiTrajectoryHelpers::trajectoryState(
+          trajectories[tips.first].multiTrajectory(), tips.second);
+      networkInput(trackID, 0) = trajState.nStates;
+      networkInput(trackID, 1) = trajState.nMeasurements;
+      networkInput(trackID, 2) = trajState.nOutliers;
+      networkInput(trackID, 3) = trajState.nHoles;
+      networkInput(trackID, 4) = trajState.NDF;
+      networkInput(trackID, 5) = (trajState.chi2Sum * 1.0) / trajState.NDF;
+      networkInput(trackID, 6) =
+          Acts::VectorHelpers::eta(parameters.momentum());
+      networkInput(trackID, 7) =
+          Acts::VectorHelpers::phi(parameters.momentum());
+      trackID++;
+    }
+  }
+  // Use the network to compute a score for all the tracks.
+  std::vector<std::vector<float>> outputTensor =
+      m_duplicateClassifier.runONNXInference(networkInput);
+  std::vector<int> goodTracks;
+  int iOut = 0;
+
+  // Loop over all the cluster and only keep the track with the highest score in
+  // each cluster
+  for (auto const& [key, val] : clusters) {
+    int bestTrackID = 0;
+    float bestTrackScore = 0;
+    for (auto const& track : val) {
+      if (outputTensor[iOut][0] > bestTrackScore) {
+        bestTrackScore = outputTensor[iOut][0];
+        bestTrackID = track;
+      }
+      iOut++;
+    }
+    goodTracks.push_back(bestTrackID);
+  }
+
+  // Create an output multitrajectory based of the good tracks
+  TrajectoriesContainer outputTrajectories;
+  outputTrajectories.reserve(goodTracks.size());
+  for (std::size_t iGoodTraj = 0; iGoodTraj < goodTracks.size(); ++iGoodTraj) {
+    const auto& iTrack = goodTracks[iGoodTraj];
+    std::pair<int, int> outputTips = trackTips.at(iTrack);
+
+    std::vector<Acts::MultiTrajectoryTraits::IndexType> tips;
+    Trajectories::IndexedParameters parameters;
+    tips.push_back(outputTips.second);
+    parameters.emplace(outputTips.second, trackParameters[iTrack]);
+
+    outputTrajectories.emplace_back(
+        trajectories[outputTips.first].multiTrajectory(), tips, parameters);
+  }
```

Comment:
I really enjoyed seeing this and particularly your talk in today's meeting.

I have a bit of a design question:
- suppose we wanted to use this ML solver in ATLAS or sPHENIX wants to use it.
- can we encapsulate this code into the `Core` or a new `Core/ML` in order to make this algorithm not more than an empty shell that does input/output, but then pushes the prepared data into a `Core/ML/DuplicateSolver` ?

Reply:
I'm not sure if we should have ONNX as a dependency of the `Core` module? If we push this code to the Onnx-Plugin, wouldn't it also be usable in ATLAS / sPHENIX?

Reply:
I don't have a strong opinion either way, in core we could have only `Core/ML` depend on Onnx. What I dislike a bit about the Plugin in that case is that if we start to have more ML base algorithm they are all going to end up in the Onnx plugin and if we start to combine multiple ML library this will also cause some some issue. We could also have a ML Plugin but at that point what is the difference between this and a `Core/ML` directory ?

Reply:
But I would argue that maybe this should be a general discussion in one of our weekly meeting on how we want to integrate ML development in Acts. And maybe for now we should just add this as is in Example ?

Reply:
For other plugins we have a split between a plugin and an examples "plugin". You need both to run it. For the python bindings, we can optionally include the ONNX examples in the python module.

Reply:
So at this point what should we do ? Do We keep the code as is for now ? Do I move it to the Onnx Plugin or another one ? Or even to Core ? 

I think it would be in everyone interest to have this merge so we can add this to the test/CI.

Reply:
I would vote for moving the parts that do not depend on the examples EDM to the plugin for now. Then it can be used without the Examples framework. However, one problem I currently see the usage of the `ActsExamples::TrajectoriesContainer`. Not sure if this could be easily replaced by the `Acts::TrackContainer` component? @paulgessinger 

If that requires some more shuffling around I would just leave it as is and refactor this at a later point.

Reply:
I am not too sure either how easy it would be to replace the `ActsExamples::TrajectoriesContainer` by `Acts::TrackContainer` ...

Reply:
Okay then I would say let's refactor this later and concentrate on the CI for now

Reply:
I think that is a story for another day as it would also effect a lot of other components. to me it makes sense to put this in for now as is. otherwise I fear we just outdate and stale it

Reply:
Works for me. As soon as we merge this I can start working on a PR to add it to the CI.
Now the question is can someone review it ?

Reply:
I probably also need to update the physmon again...

Reply:
I think we should look into porting it to `Acts::TrackContainer` but since I'm working on other parts of the chain still, let's not block this for that.